pyat/at/lattice/lattice_object.py

```diff
@@ -436,6 +454,70 @@ def particle(self, particle: Union[str, Particle]):
         else:
             self._particle = Particle(particle)
 
+    def set_fillpattern(self, bunches=1):
+        """Function to generate the filling pattern lof the ring.
+        The filling pattern is computed as:
+
+        ``bunches/numpy.sum(bunches)``
+
+        This function also generates the bunch spatial distribution
+        accessible with ``Lattice.bunch_spos``
+
+        Keyword Arguments:
+           bunches:  scalar or array of positive double or bool to define
+                     the bunch distribution.
+                     For scalar input, equidistant bunches are assumed.
+                     ``ring.harmonic_number`` has to be a multiple of
+                     ``bunches``.
+                     For array input the condition
+                     ``len(bunches)==ring.harmonic_number`` is required.
+                     (default=1, single bunch configuration).
+        """
+
+        if numpy.isscalar(bunches):
```

Comment:
I would rather test:
```
if isinstance(bunches, int);
```

Reply:
Don't we want to allow for floats as well?

Reply:
For a number of bunches ? If so, you would need to check that the decimal part is zero. I think it's not worth the effort…

Reply:
Ok but then I have to check that bunches is not a float, otherwise the error message is misleading (`len(bunches)` failing...)